src/custom/components/NotificationBanner/index.tsx

```diff
@@ -50,28 +50,22 @@ const BannerContainer = styled.div`
   justify-content: center;
 `
 export default function NotificationBanner(props: BannerProps) {
-  const [isActive, setIsActive] = useState(props.isVisible)
-  const [changeOnPropStore, setChangeOnPropStore] = useState(props.changeOnProp)
-  const { canClose = true } = props
-
+  const { id, canClose = true } = props
   const dispatch = useAppDispatch()
-  const isNotificationDismissed = useDismissNotification()
-  const noteHandleClose = () => {
-    setIsActive(false)
-    dispatch(dismissNotification(!isNotificationDismissed))
-  }
+  const isNotificationClosed = useIsNotificationClosed(id) // TODO: the notification closed state is now tied to the Affiliate state, this should generic
+  const [isActive, setIsActive] = useState(!isNotificationClosed ?? props.isVisible)
 
-  useEffect(() => {
-    if (changeOnPropStore !== props.changeOnProp) {
-      dispatch(dismissNotification(false))
-      setChangeOnPropStore(props.changeOnProp)
+  const handleClose = () => {
+    setIsActive(false)
+    if (id) {
+      dispatch(dismissNotification({ id, dismiss: true }))
```

Comment:
why `dismiss` param? do we expect anyone to invoke that with `dismiss: false`.  I would think we don't need that, and could even be a simpler signature `dismissNotification(id)`

Reply:
I left that just in case we had a scenario where we needed to toggle the visibility for an id, but apparently we managed that case differently, specially when switching networks/accounts, so I finally removed the `dismiss` parameter following your advise.